### DISCUSSION TITLE
proc-macro derive panicked

### DISCUSSION THREAD

**Discussion by Erisli** [juhaku/utoipa#446] [Q&A] [answered]
Hi, I have been trying to add ToSchema to this specific struct but I encountered the following error:

```
proc-macro derive panicked
    |
136 | #[derive(Serialize, Deserialize, Debug, Clone, ToSchema)]
    |                                                ^^^^^^^^
    |
    = help: message: TypeTreeValue from_type_paths expected at least one TypePath
```

This is the body of the struct:

```
#[derive(Serialize, Deserialize, Debug, Clone, ToSchema)]
pub struct Status{
    #[serde(rename = "Current State")]
    pub State: CurState,
    #[serde(rename = "Command")]
    pub Command: Option<String>,
    #[serde(rename = "Last Result")]
    pub LastResult: Option<Result<(), CustomError>>,
    #[serde(rename = "Last Error")]
    pub LastError: Option<CustomError>,
}
```

I am not sure where this issue comes from, all the other structures that implement ToSchema do not show any error

**Comment by juhaku** [comment, marked as answer]
I guess this error is result of this line here
> ```rust
>     pub LastResult: Option<Result<(), CustomError>>,
> ```

Currently utoipa has one limitation in types ans that is it does not support tuple types. And here I guess the utoipa mistakenly thinks that `()` is tuple type even it is unit type.

**Reply by Erisli** [reply]
Thank you for the reply! Is it ok if I create an issue for it?

**Reply by juhaku** [reply]
Yes sure,
